Improve actioncable postgres docs

sj26 opened the pull request:
Improve Action Cable PostgreSQL adapter documentation in the guides by mentioning the caveat that an extra database connection will be created from the Active Record connection pool which can exceed the pool's configured size.

I've also removed the oblique reference to future work which seems confusing, at least for the moment.

sj26 committed:
Remove vague statement about potential future

This seems confusing, and links to a stale issue.

sj26 committed:
Mention the extra postgres connection required

p8 commented:
There is a mention of the connection pool and it's size a couple of lines further down:
https://github.com/rails/rails/blob/c77bd44e1e47dd453513a4ca49132ebf2241a5fa/guides/source/action_cable_overview.md#worker-pool-configuration
Should it be mentoined there instead?

As the referenced issue hasn't been fixed after 4 years it's probably fine to delete it.
There's no need to mention all the things that might change in the guides.

sj26 commented:
I feel like this is a caveat of this particular adapter and so belongs within the adapter documentation. It also doesn't vary based on the worker pool configuration, and is separately from the database connection pool, so feels a bit orthogonal. But it could be mentioned in that lower part if you really thought it better.

p8 commented:
In that case should the text mention the adapter?
> **PostgreSQL adapter** will take an extra connection...

sj26 commented:
That's feels very repetitive, given it's within a heading for the adapter, and amongst many mentions of the adapter:

![image](https://user-images.githubusercontent.com/14028/93288980-77286780-f820-11ea-8e8c-3ad800766ba2.png)

sj26 commented:
Perhaps it should be merged into a single paragraph, on second look.

sj26 committed:
Combine into a single paragraph

sj26 committed:
Remove expain-y advice

This is implied by highlighting the connection being held outside the
pool limits.

sj26 committed:
Align terminology

Most other places simply talk about the 'server process'
